addons/website/tools.py

```diff
@@ -169,6 +169,16 @@ def text_from_html(html_fragment, collapse_whitespace=False):
     """
     # lxml requires one single root element
     tree = etree.fromstring('<p>%s</p>' % html_fragment, etree.XMLParser(recover=True))
+
+    # Remove scripts or other technical elements that should not be converted
+    # into text.
+    xpath_filters = [
+        '//script',
+        '//*[@class="css_non_editable_mode_hidden"]'
+    ]
+    for xpath_filter in xpath_filters:
+        for element in tree.xpath(xpath_filter): element.getparent().remove(element)
```

Comment:
With the `recover` parser attribute set to True, if the lxml parser recovers in a way that leaves certain nodes disconnected from a parent, those nodes might have None as their parent.
I propose:
```python
for element in tree.xpath(xpath_filter):
    parent = element.getparent()
         if parent is not None:
               parent.remove(element)
```